hare/hare.go

```diff
@@ -241,6 +241,9 @@ func (h *Hare) collectOutput(ctx context.Context, output TerminationOutput) erro
 			return fmt.Errorf("hare save block: %w", err)
 		} else {
 			hareOutput = block.ID()
+			if err = h.pdb.DelProposals(pids); err != nil {
```

Comment:
we should not delete it right away for two reasons
- some peers may be trying to fetch proposals from the node
- debugging

perhaps for now we can keep a grace period say one epoch?